barman/retention_policies.py

```diff
@@ -126,6 +126,31 @@ def to_json(self):
         """
         return "%s %s %s" % (self.mode, self.value, self.unit)
 
+    def _propagate_retention_status_to_children(self, backup_info, report, ret_status):
+        """
+        Propagate retention status to all backups in the tree.
+
+        .. note::
+            This has a side-effect. It modifies or add data to *report* dict.
+
+        :param barman.infofile.BackupInfo backup_info: The object we want to
+            propagate the RETENTION STATUS from.
+        :param dict[str, str] report: The report data structure to be modified.
+            Each key is the ID of a backup, and its value is the retention status
+            of that backup.
+        :param str ret_status:  The status of the backup according to retention
+            policies
+        """
+        if ret_status in (BackupInfo.KEEP_STANDALONE, BackupInfo.KEEP_FULL):
```

Comment:
```suggestion
        # As KEEP status doesn't make sense for incremental backups, we simply
        # set them as VALID if their root full backup has KEEP annotation
        if ret_status in (BackupInfo.KEEP_STANDALONE, BackupInfo.KEEP_FULL):
```

Maybe it's worth it a comment.